File: SoftUni Lessons/Python Development/Python Basics April 2019/Past Exams and Problems/7 - 9 and 10 March 2019/06. Basketball Tournament.py
tournament_name = None
count_games = 0
points_1 = 0
points_2 = 0
games_won = 0
games_lost = 0

while True:
    tournament_name = input()
    if tournament_name == "End of tournaments":
        break
    count_games = int(input())
    for game in range(1, count_games + 1):
        points_1 = int(input())
        points_2 = int(input())
        if points_1 > points_2:
            print(f"Game {game} of tournament {tournament_name}: win with {points_1 - points_2} points.")
            games_won += 1
        else:
            print(f"Game {game} of tournament {tournament_name}: lost with {points_2 - points_1} points.")
            games_lost += 1

total_games = games_won + games_lost

print(f"{(games_won / total_games * 100):.2f}% matches win")
print(f"{(games_lost / total_games * 100):.2f}% matches lost")

File: SoftUni Lessons/JavaScript Development/JavaScript Basics March 2020/21. МИНАЛИ ИЗПИТИ/03. Online Exam - 9 and 10 March 2019/06. Basketball Tournament.js
function basketballTournament(input) {
    // Mask
    let tournamentName = undefined;
    let countGames = 0;
    let pointsOne = 0;
    let pointsTwo = 0;
    let gamesWon = 0;
    let gamesLost = 0;

    while ('I say go!') {
        tournamentName = input.shift();
        if (tournamentName == "End of tournaments") {break;}
        countGames = Number(input.shift());
        for (game = 1; game <= countGames; game +=1) {
            let pointsOne = Number(input.shift());
            let pointsTwo = Number(input.shift());
            if (pointsOne > pointsTwo) {
                console.log(`Game ${game} of tournament ${tournamentName}: win with ${pointsOne - pointsTwo} points.`);
                gamesWon += 1;
            } else {
                console.log(`Game ${game} of tournament ${tournamentName}: lost with ${pointsTwo - pointsOne} points.`);
                gamesLost += 1;
            }
        }
        
    }

    let totalGames = gamesWon + gamesLost;

    console.log(`${(gamesWon / totalGames * 100).toFixed(2)}% matches win`);
    console.log(`${(gamesLost / totalGames * 100).toFixed(2)}% matches lost`);
}


basketballTournament(['Dunkers', 2, 75, 65, 56, 73, 'Fire Girls', 3, 67, 34, 83, 98, 66, 45, 'End of tournaments']);  
// Should return:
// Game 1 of tournament Dunkers: win with 10 points.
// Game 2 of tournament Dunkers: lost with 17 points.
// Game 1 of tournament Fire Girls: win with 33 points.
// Game 2 of tournament Fire Girls: lost with 15 points.
// Game 3 of tournament Fire Girls: win with 21 points.
// 60.00% matches win
// 40.00% matches lost


basketballTournament(['Ballers', 3, 87, 63, 56, 65, 75, 64, 'Sharks', 4, 64, 76, 65, 86, 68, 99, 45, 78, 'End of tournaments ']);  
// Should return:
// Game 1 of tournament Ballers: win with 24 points.
// Game 2 of tournament Ballers: lost with 9 points.
// Game 3 of tournament Ballers: win with 11 points.
// Game 1 of tournament Sharks: lost with 12 points.
// Game 2 of tournament Sharks: lost with 21 points.
// Game 3 of tournament Sharks: lost with 31 points.
// Game 4 of tournament Sharks: lost with 33 points.
// 28.57% matches win
// 71.43% matches lost
